### PR TITLE
Enhancement/9568 Improve rendering time for SiwG Setup CTA

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #9568 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @jimmymadon – I made a few changes in trying to work out the test instabilities which I elaborated on in our DM, just a few outstanding points to address.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget/index.js [hunk 1]
```diff
@@ -110,6 +113,7 @@ function AudienceSegmentationSetupCTAWidget( { Widget, WidgetNull } ) {
 			onSuccess: () => {
 				invalidateResolution( 'getQueuedNotifications', [
 					viewContext,
+					NOTIFICATION_GROUPS.DEFAULT,
```
[the remark this answers is not in the document]
Comment (jimmymadon) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget/index.js:116]: The notification to render is actually the "Audience Segmentation Setup Success Subtle Notification" which renders when the setup is successful. This Setup CTA Component itself is not yet 'refactored' to use the new Notifications infrastructure. 

On a side note, this is still a "hack" which should be removed with #9453

**Comment by jimmymadon** [comment]
@aaemnnosttv Thanks for the great clean up here and for introducing this new pattern to focus on testing `checkRequirements` alone. I've added a JS test for the groupID filtering.

The VRT failure seems to be unrelated to this PR.

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks @jimmymadon !

**Pull request merged**